__tests__/app.test.js

```diff
@@ -173,6 +172,177 @@ describe('GET /api/articles/:article_id/comments', () => {
+    });
+    test('if there is article_id is not valid responds with 400', () => {
+        const newComment = {
+        username: "rogersop",
+        body: 'this is test for a username missing from a comment'
+    }
+    
+    return request(app)
+    .post(`/api/articles/notID/comments`)
+    .send(newComment)
+    .expect(400)
+    .then(({body}) => {
+        expect(body.message).toBe('Invalid article_id' )
+        })
+    });  
+});
+
+// 8
+describe('PATCH /api/articles/:article_id', () => {
+    test('200: responds with updated article', () => {
+        const articleUpdate = { inc_votes : 1 }
+        return request(app)
+        .patch('/api/articles/1')
+        .send(articleUpdate)
+        .expect(200)
+        .then((response) => {
+            const {article} = response.body;
+            console.log("article  ", article)
+            expect(article).toEqual({
+                article_id: 1,
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 101,
+                article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+
+            })
+        })
+    });
+    test('200: responds with updated article', () => {
+        const articleUpdate = { inc_votes : -1 }
+        return request(app)
+        .patch('/api/articles/1')
+        .send(articleUpdate)
+        .expect(200)
+        .then((response) => {
+            const {article} = response.body;
+            // console.log("article  ", article)
+            expect(article).toEqual({
+                article_id: 1,
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 99,
+                article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+
+            })
+        })
+    });
+
+    test('404: article doesnt exist', () => {
+        const articleUpdate = { inc_votes : 1 }
+        return request(app)
+        .patch('/api/articles/456454')
+        .send(articleUpdate)
+        .expect(404)
+        .then(({body}) => {
+             
+            expect(body.message).toEqual('Article does not exist')   
+        })
+    });
+    test('400: votes less than 0', () => {
+        const articleUpdate = { inc_votes : -5465 }
+        return request(app)
+        .patch('/api/articles/1')
+        .send(articleUpdate)
+        .expect(400)
+        .then(({body}) => {
+             console.log(body.message)
+            expect(body.message).toEqual('Votes cannot be negative')   
+        })
+    });
+
+})
```

Comment:
Your post tests look great! Big improvement. 

I would be careful with these patch tests, on line 300 you're expecting a 200 and inc_votes = -1, then on line 333 you're saying inc_votes can't be negative or you'll get an error. I think it's beneficial if inc_votes _can_ be negative, because if you accidentally voted, you'd want to be able to "unvote" 

Also missing a 400 test if inc_votes is not a number & 400 test invalid id eg. /api/articles/notanid 👍

Include these tests and submit another PR for review!